docs/recipes/automate-release-workflow.md

```diff
@@ -1,79 +1,94 @@
-// command argument whether you are doing a 'major', 'minor' or a 'patch' change.
-  return gulp.src(['./bower.json', './package.json'])
-    .pipe(bump({type: "patch"}).on('error', log.error))
-    .pipe(gulp.dest('./'));
-});
-
-gulp.task('commit-changes', function () {
-  return gulp.src('.')
-    .pipe(git.add())
-    .pipe(git.commit('[Prerelease] Bumped version number'));
-});
-
-gulp.task('push-changes', function (done) {
-  git.push('origin', 'master', done);
-});
-
-gulp.task('create-new-tag', function (done) {
-  var version = getPackageJsonVersion();
-  git.tag(version, 'Created Tag for version: ' + version, function (error) {
-    if (error) {
-      return done(error);
-    }
-    git.push('origin', 'master', {args: '--tags'}, done);
+The recipe below bumps the project version, commits the changes to git and creates a new GitHub release.
+
+For publishing a GitHub release you'll need to [create a personal access token](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token) and add it to your project. However, we don't want to commit it, so we'll use [`dotenv-safe`](https://www.npmjs.com/package/dotenv-safe) to load it from a git-ignored `.env` file.
+
+Create an `.env.example` file to define required environment variables:
+
+```
+GH_TOKEN=
+```
+
+We'll commit this, so leave it empty. Next, duplicate this file, rename it to `.env` and add your GitHub token:
+
+```
+GH_TOKEN=ff34885...
+```
+
+Finally, add `.env` to your `.gitignore`.
+
+Moving on, install all the necessary dependencies for this recipe:
+
+```sh
+npm install --save-dev conventional-recommended-bump conventional-changelog-cli conventional-github-releaser dotenv-safe execa pify
+```
+
+Finally, `gulpfile.js`:
+
+``` js
+const gulp = require('gulp');
+const conventionalRecommendedBump = require('conventional-recommended-bump');
+const conventionalGithubReleaser = require('conventional-github-releaser');
+const execa = require('execa');
+const fs = require('fs/promises');
+const pify = require('pify');
+const dotenv = require('dotenv-safe');
+
+// load environment variables
+dotenv.config();
+
+// Conventional Changelog preset
+const preset = 'angular';
+// print output of commands into the terminal
+const stdio = 'inherit';
+
+async function bumpVersion() {
+  // get recommended version bump based on commits
+  const { releaseType } = await pify(conventionalRecommendedBump)({ preset });
+  // bump version without committing and tagging
+  await execa('npm', ['version', releaseType, '--no-git-tag-version'], {
+    stdio,
   });
+}
+
+async function changelog() {
+  await execa(
+    'npx',
+    [
+      'conventional-changelog',
+      '--preset',
+      preset,
+      '--infile',
+      'CHANGELOG.md',
+      '--same-file',
+    ],
+    { stdio }
+  );
+}
+
+async function commitTagPush() {
+  const { version } = JSON.parse(await fs.readFile('package.json'));
```

Comment:
Even though we could get away with it here, I feel uncomfortable about using it because `require` gets cached, so if we happen to `require` `package.json` multiple times in the workflow, we wouldn't get the current value, but instead the value since the last time we called `require`. So the fact that it won't be a problem here is just a coincidence. I added a comment explaining this, so that people know that they can use it if they want to.

If you think this is an overkill for a recipe, I'll use `require` instead.

Reply:
This is a good reasoning and I agree with you!